### ISSUE TITLE
Http : Caller's remote-address.host property has never been set 

### ISSUE THREAD

**Issue by VirajSalaka** [ballerina-platform/ballerina-lang#22283]
**Description:**
http:Caller's remoteAddress.host property is not set at all. 

Previously, it was doing a reverse dns lookup and that caused a significant delay (~15s) when the call is from a remote network. But IMO, Ip can be added instead of the hostname. 
https://github.com/ballerina-platform/ballerina-lang/issues/20085

**Steps to reproduce:**
Make a call from remote network and observe the property.

**Affected Versions:**
1.1.4

**OS, DB, other environment details and versions:**
MacOS Mojave

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by VirajSalaka** [comment]
@chamil321 Please mention if there is any information is missing.

**Comment by VirajSalaka** [comment]
@shafreenAnfar Appreciate if you can take a look as well :) .

**Comment by chamil321** [comment]
Shall we use the following method which avoids reverse lookup to set the remote IP?


> public final String **getHostString()**

_Returns the hostname, or the String form of the address if it doesn't have a hostname (it was created using a literal). This has the benefit of not attempting a reverse lookup._


https://docs.oracle.com/javase/7/docs/api/java/net/InetSocketAddress.html#getHostString()

**Comment by chamil321** [comment]
After discussing the solution @shafreenAnfar, A function will be introduced along with the aforementioned  change to get the hostname by triggering the reverse lookup. 

So that the user doesn't lose the info and most importantly user is aware.

**Comment by chamil321** [comment]
The function will be added to caller
```ballerina
string? remoteHostName = caller.getRemoteHostName();
```

Definition : 
```ballerina
# Gets the hostname from the remote address. This method may trigger DNS reverse lookup if the address was created
# with a literal IP address.
#
# + return - the hostname part of the address or nil if it is unresolved
public function getRemoteHostName() returns string? {}
```

**Comment by chamil321** [comment]
Fixed by #22569, hence closing the issue
